File: 2024_12/leetcode2024-12-22.py
class Solution:
    def getKth(self, lo: int, hi: int, k: int) -> int:
        def getVal(x):
            cnt = 0
            while x>1:
                if x&1:
                    x = x*3+1
                else:
                    x //= 2
                cnt += 1
            return cnt
        
        vals = [(i,getVal(i)) for i in range(lo, hi+1)]
        vals.sort(key = lambda x:x[1])
        return vals[k-1][0]

if __name__ == "__main__":
    s = Solution()
    lo, hi, k = 12, 15, 2
    print(s.getKth(lo, hi, k))

File: leetcode2024/2024_12/leetcode2024-12-22.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
private:
    int getVal(int x){
        int cnt = 0;
        while (x>1){
            if (x&1){
                x = x*3+1;
            }
            else{
                x /= 2;
            }
            cnt += 1;
        }
        return cnt;
    }
public:
    int getKth(int lo, int hi, int k) {
        vector<vector<int>> vals;
        for (int i=lo; i<=hi; i++){
            vals.push_back({getVal(i), i});
        }
        sort(vals.begin(), vals.end());
        return vals[k-1][1];
    }
};

int main(){
    int lo = 12, hi = 15, k = 2;
    Solution s;
    cout << s.getKth(lo, hi, k) << endl;
}
